apps/feedback/tests/api_tests.py

```diff
@@ -224,6 +225,33 @@ def test_users_cannot_answer_without_filling_all_fields(self):
 
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
+    def test_answer_only_filling_required_fields(self):
+        rating_question = self.create_rating_question(required=True)
+        self.create_text_question(required=False)
+        relation = self.create_feedback_relation(user=self.user)
+        response = self.client.post(
+            self.get_submit_url(relation),
+            {"rating_answers": [{"question": rating_question.id, "answer": 1}]},
+            **self.headers,
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_non_required_field_with_blank_answer(self):
+        rating_question = self.create_rating_question(required=True)
+        text_question = self.create_text_question(required=False)
+        relation = self.create_feedback_relation(user=self.user)
+        response = self.client.post(
+            self.get_submit_url(relation),
+            {
+                "rating_answers": [{"question": rating_question.id, "answer": 1}],
+                "text_answers": [{"question": text_question.id, "answer": ""}],
+            },
+            **self.headers,
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
Denne forstod jeg ikke helt, skal "non-required fields with black answer" gi 400 Bad Request? Burde ikke non-required fields kunne tillate et tomt svar?

Reply:
Du har egentlig rett der, og jeg er litt usikker på hva som er den beste måten å håndere det på.

Slik det virker nå må du unngå å sende inn et svar på spørsmålet i API-kallet, fordi valideringen til objektene sier at et svar ikke skal være blankt. Det er jo egentlig ubrukelig informasjon å lagre et blank svar, og da er det kanskje også ubrukelig at det skal sendes inn?

Vi har sånn sett to valg:
1. Tillate at du sender tomme svar, hvor vi på en eller annen måte må skru av validering av blanke felt for modellen hvis det feltet er satt, og så filtrere utsvarene som er blanke fra dataen.

2. Ikke tillate at det sendes noe data på spørsmålet når svaret er blankt, som krever at du filtrerer det ut i front-end og sikkert kan være litt forvirrende?

Reply:
Jeg har sett en del på det, og det virker som en mye større jobb å filtrere ut svarene i back-end i dette tilfellet. Tror det går helt greit å bare filtrere ut spørsmål som ikke er besvart i front-end?

Reply:
Jeg syntes det er greit at det er sånn det fungerer, men skulle gjerne sett at det ble dokumentert på et fornuftig sted, kanskje i `/api/v1/`?